libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/NotificationCommon.java

```diff
@@ -97,19 +98,27 @@ public Notification createNotification(final String recipient,
     }
 
     void addDomainRoleRecipients(Notification notification, final String domainName, final String roleName) {
-        Set<String> domainRoleMembers = domainRoleMembersFetcher.getDomainRoleMembers(domainName, roleName);
-        if (domainRoleMembers == null || domainRoleMembers.isEmpty()) {
+        try {
+            Set<String> domainRoleMembers = domainRoleMembersFetcher.getDomainRoleMembers(domainName, roleName);
+            if (domainRoleMembers == null || domainRoleMembers.isEmpty()) {
+                return;
+            }
+
+            notification.getRecipients().addAll(domainRoleMembers);
+        } catch (ResourceException ex) {
+            LOGGER.error("Error getting domain role members ", ex);
             return;
         }
-
-        notification.getRecipients().addAll(domainRoleMembers);
     }
 
     void addNotificationRecipient(Notification notification, final String recipient, boolean ignoreService) {
 
-        int idx = recipient.indexOf(AuthorityConsts.ROLE_SEP);
-        if (idx != -1) {
-            addDomainRoleRecipients(notification, recipient.substring(0, idx), recipient);
+        int roleDomainIndex = recipient.indexOf(AuthorityConsts.ROLE_SEP);
+        int groupDomainIndex = recipient.indexOf(AuthorityConsts.GROUP_SEP);
+        if (roleDomainIndex != -1) {
+            addDomainRoleRecipients(notification, recipient.substring(0, roleDomainIndex), recipient);
+        } else if (groupDomainIndex != -1) {
```

Comment:
Domain admins will get the notification but I don't think it's reasonable to send every member of a group notification if it expires on a specific role.